syntax = "proto3";
import "Msg_Base.proto";
message ResultObject{
    uint64 ObjectID = 1;
    int32  ActorID = 2;
    int32  ObjectType = 3;
    int32  Result = 4;
    int32  Kill = 5;
    int32  Death = 6;
    int32  Heal = 7;
    int32  Damage = 8;
}

message ResultPlayer {
    uint64 ObjectID = 1;
    int32  ActorID = 2;
    int32  ObjectType = 3;
    int32  Result = 4;
    int32  Kill = 5;
    int32  Death = 6;
    int32  Heal = 7;
    int32  Damage = 8;
}

message BattleResultNty {
    int32   ServerID    = 1;
    uint32  CopyGuid    = 2;
    int32   CopyID      = 3;
    int32   CopyType    = 4;
    uint64  LastTime    = 5;
    repeated ResultPlayer PlayerList = 6;   //玩家列表
    repeated ResultObject ObjectList = 7;   //宠物，伙伴之类的
}

message MainCopyReq {
    int32   CopyID  = 1;  //副本类型
}

message MainCopyAck {
    uint32   RetCode = 1;
}

message MainCopyResultNty {
    uint64   RoleID             = 1; //角色ID
    int32    CopyID             = 2; //副本ID
    int32    CopyResult         = 3; //胜负结果
    int32    StarNum            = 4; //星数
    uint64   LastTime           = 5; //战斗持续时间
    repeated ItemData ItemList  = 6; //获得道具
}

message Msg_CreateRoomReq{
	uint64  RoleID = 1;
	uint32 	CopyID = 2;
}

message Msg_CreateRoomAck{
	uint32   RetCode = 1;
	uint64   RoomID  = 2;
}

message Msg_JoinRoomReq{
	uint64  RoleID = 1;
	uint64  RoomID  = 2;
}

message Msg_JoinRoomAck{
	uint32   RetCode = 1;
	uint64   RoomID  = 2;
}

message Msg_KickRoomReq{
	uint64  RoleID = 1;
	uint64  RoomID  = 2;
}

message Msg_KickRoomAck{
	uint32   RetCode = 1;
}

message Msg_LeaveRoomReq{
	uint64  RoleID = 1;
	uint64  RoomID  = 2;
}

message Msg_LeaveRoomAck{
	uint32   RetCode = 1;
}

message Msg_StartRoomReq{
	uint64  RoleID = 1;
	uint64  RoomID  = 2;
}

message Msg_StartRoomAck{
	uint32   RetCode = 1;
}

message Msg_RoomPlayerInfo{
	uint64 RoleID 			= 1;    //角色ID
	int32  Level 			= 2;	//等级
	string Name 			= 3;	//名字
	int32     ActorID       = 4;	//Actor ID
    int32     MountID       = 5;	//坐骑ID
    repeated int32 Equips   = 6;	//装备
}

message Msg_BroadRoomNotify{
	repeated Msg_RoomPlayerInfo PlayerList = 1;
}
